delta/service/src/main/scala/ch/epfl/bluebrain/nexus/delta/service/resolvers/ResolversImpl.scala

```diff
@@ -0,0 +1,250 @@
+        UnscoredSearchResults(
+          results.size.toLong,
+          results.map(UnscoredResultEntry(_)).slice(pagination.from, pagination.from + pagination.size)
+        )
+      }
+      .named("listResolvers", moduleType)
+
+  override def events(offset: Offset): fs2.Stream[Task, Envelope[ResolverEvent]] =
+    eventLog.eventsByTag(moduleType, offset)
+
+  private def eval(cmd: ResolverCommand, project: Project): IO[ResolverRejection, ResolverResource] =
+    for {
+      evaluationResult <- agg.evaluate(identifier(cmd.project, cmd.id), cmd).mapError(_.value)
+      (am, base)        = project.apiMappings -> project.base
+      resource         <-
+        IO.fromOption(evaluationResult.state.toResource(am, base), UnexpectedInitialState(cmd.id, project.ref))
+      _                <- index.put(cmd.project -> cmd.id, resource)
+    } yield resource
+
+  private def identifier(projectRef: ProjectRef, id: Iri): String =
+    s"${projectRef}_$id"
+}
+
+object ResolversImpl {
+
+  type ResolversAggregate = Aggregate[String, ResolverState, ResolverCommand, ResolverEvent, ResolverRejection]
+  type ResolversCache     = KeyValueStore[(ProjectRef, Iri), ResolverResource]
+
+  private val logger: Logger = Logger[ResolversImpl]
+
+  /**
+    * Creates a new resolvers cache.
+    */
+  private def cache(config: ResolversConfig)(implicit as: ActorSystem[Nothing]): ResolversCache = {
+    implicit val cfg: KeyValueStoreConfig       = config.keyValueStore
+    val clock: (Long, ResolverResource) => Long = (_, resource) => resource.rev
+    KeyValueStore.distributed(moduleType, clock)
+  }
+
+  private def startIndexing(
+      config: ResolversConfig,
+      eventLog: EventLog[Envelope[ResolverEvent]],
+      index: ResolversCache,
+      resolvers: Resolvers
+  )(implicit as: ActorSystem[Nothing], sc: Scheduler) =
+    StreamSupervisor.runAsSingleton(
+      "ResolverIndex",
+      streamTask = Task.delay(
+        eventLog
+          .eventsByTag(moduleType, Offset.noOffset)
+          .mapAsync(config.indexing.concurrency)(envelope =>
+            resolvers
+              .fetch(IriSegment(envelope.event.id), envelope.event.project)
+              .flatMap {
+                case Some(res) => index.put(res.value.project -> res.value.id, res)
+                case None      => UIO.unit
+              }
+              .onErrorHandle(_ => ())
+          )
+      ),
+      retryStrategy = RetryStrategy(
+        config.indexing.retry,
+        _ => true,
+        RetryStrategy.logError(logger, "resolvers indexing")
+      )
+    )
+
+  private def aggregate(config: AggregateConfig)(implicit as: ActorSystem[Nothing], clock: Clock[UIO]) = {
+    val definition = PersistentEventDefinition(
+      entityType = moduleType,
+      initialState = Initial,
+      next = Resolvers.next,
+      evaluate = Resolvers.evaluate,
+      tagger = (_: ResolverEvent) => Set(moduleType),
```

Comment:
If we want to support, as we discussed this morning:
- v1/resolvers/{org}/{project}/event
- v1/resolvers/{org}/event
- v1/resources/{org}/{project}/event (including resolvers)
- v1/resources/{org}/event (including resolvers)
...we will also need the following tags:
```scala
tagger = (ev: ResolverEvent) =>
        Set(
          moduleType,
          s"${Projects.moduleType}=${ev.project}",
          s"${Organizations.moduleType}=${ev.project.organization}"
        )
```